DigitalLearningSolutions.Web/Controllers/Register/RegisterDelegateByCentreController.cs

```diff
@@ -0,0 +1,195 @@
+namespace DigitalLearningSolutions.Web.Controllers.Register
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using DigitalLearningSolutions.Data.DataServices;
+    using DigitalLearningSolutions.Data.Services;
+    using DigitalLearningSolutions.Web.Extensions;
+    using DigitalLearningSolutions.Web.Helpers;
+    using DigitalLearningSolutions.Web.Models;
+    using DigitalLearningSolutions.Web.ServiceFilter;
+    using DigitalLearningSolutions.Web.ViewModels.Common;
+    using DigitalLearningSolutions.Web.ViewModels.Register;
+    using Microsoft.AspNetCore.Authorization;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.AspNetCore.Mvc;
+
+    [Authorize(Policy = CustomPolicies.UserCentreAdmin)]
+    [Route("/TrackingSystem/Delegates/Register")]
+    public class RegisterDelegateByCentreController : Controller
+    {
+        private const string CookieName = "DelegateRegistrationByCentreData";
+        private readonly CustomPromptHelper customPromptHelper;
+        private readonly IJobGroupsDataService jobGroupsDataService;
+        private readonly IUserService userService;
+
+        public RegisterDelegateByCentreController(
+            IJobGroupsDataService jobGroupsDataService,
+            IUserService userService,
+            CustomPromptHelper customPromptHelper
+        )
+        {
+            this.jobGroupsDataService = jobGroupsDataService;
+            this.userService = userService;
+            this.customPromptHelper = customPromptHelper;
+        }
+
+        public IActionResult Index()
+        {
+            var centreId = User.GetCentreId();
+
+            SetCentreDelegateRegistrationData(centreId);
+
+            return RedirectToAction("PersonalInformation");
+        }
+
+        [ServiceFilter(typeof(RedirectEmptySessionData<DelegateRegistrationByCentreData>))]
+        [HttpGet]
+        [Route("PersonalInformation")]

```

Comment:
A lot of these methods have these Route attributes that just restate the name of the method. Can these be removed safely, or does it mess everything up?

Reply:
It messes everything up. I think if there's a Route on the class then you need to add Routes to the methods otherwise it won't know which method to use when navigating to the class Route.

Reply:
Okay I've cleaned it up a bit by adding `{action}` to the class Route, and only specifying the root Route for the Index action. The other action routes get inferred automatically.